pysynchronizer/objects.py

```diff
@@ -0,0 +1,188 @@
+#
+# Copyright (c) 2021 Daniel P. Smith, Apertus Solutions LLC
+# 
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+
+from .utils import is_valid_uuid, path_to_uuid, uuid_to_path
+from .errors import ConnectionError
+from .storage import Storage
+from .oxt_dbus import OXTDBusApi
+
+class XenMgr:
+    def __init__(self):
+        try:
+            self.xenmgr = OXTDBusApi.open_xenmgr()
+            self.host = OXTDBusApi.open_host()
+        except Exception as err:
+           raise ConnectionError('Failed to connect to XenMgr service') from err
+
+    def get_current_release_and_build(self):
+        """Get current XC release and build number from xenmgr"""
+        build_info = self.host.get_property('build-info')
+        return build_info['release'], build_info['build']
+
+    def vm_by_name(self):
+        """Returns a dictionary of name:uuid pairs"""
+        d = {}
+
+        for path in self.xenmgr.list_vms():
+            vm_uuid = path_to_uuid(path)
+            name = OXTDBusApi.open_vm(path).get_property("name")
+            d[name] = vm_uuid
+
+        return d
+
+    def vm_by_uuid(self):
+        """Returns a dictionary of uuid:name pairs"""
+        d = {}
+
+        for path in self.xenmgr.list_vms():
+            vm_uuid = path_to_uuid(path)
+            name = OXTDBusApi.open_vm(path).get_property("name")
+            d[vm_uuid] = name
+
+        return d
+
+    def find_vm(self, ref):
+        """Takes a name or uuid and return a uuid if VM is found"""
+        if is_valid_uuid(ref):
+            vms = self.vm_by_uuid()
+            if ref in vms:
+                return ref
+            else:
```

Comment:
Unneeded, we'll return None at the outer scope already.

Reply:
yep, will fix